Cloud Console Insights Page (#16301)
cockroachdb/docs: _includes/v23.1/ui/insights.md

> ## Workload Insights tab

Reviewer: this page appears to be the same across v22.2 and v23.1?  at least I couldn't discern any big differences.  If they are the same, see my suggestions on v22.2 version of this page - from looking through, all the same suggestions would apply (and please feel free to ignore if I'm missing some different details that would make them not apply of course!)

Reviewer: ah ok I took a closer look.  at least from what I can see there are 3 areas of difference, but they don't really change my suggestions which are basically to add as many links to things as possible :-)

    - **Drop Unused Index**: An **Index** field displays the name of the index to drop; and a **Description** field displays the reason for dropping the index.

[Admin users]({{ link_prefix }}security-reference/authorization.html#admin-role) will see an action button in the final column, which will execute the SQL statement suggested by the schema insight, for example "Create Index". Upon clicking the action button, a confirmation dialog displays a warning about the cost of [online schema changes]({{ link_prefix }}online-schema-changes.html) and the option to copy the SQL statement for later execution in a SQL client.

## Configuration

You can configure the behavior of insights using the following [cluster settings]({{ link_prefix }}cluster-settings.html).

### Workload insights settings

You can configure [**Workload Insights**](#workload-insights-tab) with the following [{{ link_prefix }}cluster settings]({{ link_prefix }}cluster-settings.html):

> | Setting                                                                | Default value | Description                                                                                                                                                                                   | Where used                           |

Reviewer: Same suggestion here as the other page for the tables: adding links, etc.

## Workload Insights tab

The **Workload Insights** tab displays insights related to transaction and statement executions.

### Transaction Executions view

{% if page.cloud != true -%}
To display this view, click **Insights** in the left-hand navigation of the DB Console and select **Workload Insights > Transaction Executions**.
{% endif -%}
{% if page.cloud == true -%}
{{site.data.alerts.callout_info}}

> The **Transaction Executions** view is currently unavailable for {{ site.data.products.serverless }} clusters.

Reviewer 2: ```suggestion
The **Transaction Executions** view is currently unavailable for {{ site.data.products.serverless }} clusters.
```

This isn't (won't be) true for 23.1.

- **Statement Fingerprint ID**: The statement fingerprint ID of the latest statement execution.
- **Statement Execution**: The [statement fingerprint]({{ link_prefix }}ui-statements-page.html#sql-statement-fingerprints) of the latest statement execution.
- **Insights**: The insight for the statement execution.
  - **High Contention**: The statement execution experienced high contention time according to the threshold set in `sql.insights.latency_threshold`.
  - **High Retry Count**: The statement execution experienced a high number of retries according to the threshold set in `sql.insights.high_retry_count.threshold`.
  - **Suboptimal Plan**: The statement execution has resulted in one or more [index recommendations](#schema-insights-tab) that would improve the plan.
  - **Failed**: The statement execution failed.
  - **Slow Execution**: The statement experienced slow execution. Depending on the settings in [Configuration](#configuration), either of the following conditions trigger this insight:

      - Execution time is greater than the value of `sql.insights.latency_threshold`.

>       - Anomaly detection is enabled (`sql.insights.anomaly_detection.enabled`), execution time is greater than the value of `sql.insights.anomaly_detection.latency_threshold`, and [execution latency]({{ link_prefix }}ui-sql-dashboard.html#kv-execution-latency-99th-percentile) is `> p99 && > 2*p50`. For details, see [Detect slow executions](#detect-slow-executions).

Reviewer 2: ```suggestion
      - Anomaly detection is enabled (`sql.insights.anomaly_detection.enabled`), execution time is greater than the value of `sql.insights.anomaly_detection.latency_threshold`, and [execution latency]({{ link_prefix }}ui-sql-dashboard.html#kv-execution-latency-99th-percentile) is greater than the p99 latency and more than double the median latency. For details, see [Detect slow executions](#detect-slow-executions).
```

Reviewer 2: (Same for all instances of  `> p99 && > 2*p50`.)

- **Start Time**: The timestamp when the statement execution started.
- **End Time**: The timestamp when the statement execution ended.
- **Elapsed Time**: The time that elapsed during statement execution.
- **Rows Read**: The total number of rows read.
- **Rows Written**: The total number of rows written.
- **Transaction Priority**: The [priority]({{ link_prefix }}transactions.html#set-transaction-priority) of the transaction for the statement execution.
- **Full Scan**: Whether the execution performed a full scan of the table.
- **Session ID**: The ID of the [session]({{ link_prefix }}ui-sessions-page.html) the statement was executed from.
- **Transaction Fingerprint ID**: The ID of the transaction fingerprint for the statement execution.
- **Transaction Execution ID**: The ID of the transaction execution for the statement execution.
- **Statement Fingerprint ID**: The fingerprint ID of the statement fingerprint for the statement execution.

## Schema Insights tab

> To display this view, click **Insights** in the left-hand navigation of the DB Console and select **Schema Insights**.

Reviewer 2: ```suggestion
```

(duplicated)
